models.py

```diff
@@ -210,85 +210,166 @@ def empty_order(self):
         Remove all the items from the order.
         Delete all OrderItem related to this order and reset the total_price
         value to 0.
-
         """
 
         self.total_price = 0
         OrderItem.delete().where(OrderItem.order == self).execute()
         self.save()
         return self
 
-    def add_item(self, item, quantity=1):
+    @staticmethod
+    def create_order(user, address, items):
+        """
+        Create an Order and respective OrderItems. OrderItems are created
+        in a single query as well as the Order. It also updates Items'
+        availability.
+        :param User user:
+        :param Address address:
+        :param dict items: {<Item>: <quantity:int>}
         """
-        Add one item to the order.
-        Creates one OrderItem row if the item is not present in the order yet,
-        or increasing the count of the existing OrderItem. It also updates the
-        item availability counter and raise InsufficientAvailability if
-        quantity is less than item availability.
+        total_price = sum([
+            item.price * quantity for item, quantity in items.items()])
```

Comment:
oppure, in modo migliore:
```
sum(item.price * quantity for item, quantity in items.items())
```
non hai bisogno di creare una lista intermedia solo per passarla alla sum, puoi passargli un generator expression